src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx

```diff
@@ -101,9 +102,69 @@ const RuleEditor = createReactClass({
     Promise.all(promises).then(([environments, rule]) => {
       this.setState({environments, rule});
     });
-  },
+  }
+
+  pollHandler = async quitTime => {
+    if (Date.now() > quitTime) {
+      addErrorMessage(t('Looking for that channel took too long :('));
+      this.setState({loading: false});
```

Comment:
@MeredithAnya Nit: you could return after the `setState` call then avoid having to have an `else` condition and having things tabbed over.